### PR TITLE
Make changes in xfstest for redfs run

### PR DESCRIPTION
**Pull request by d-hans**
With these changes passthrough_hp completed run on xfstest without any issue in kernel.
(Testing included redfs module, red-libfuse on linux-5.19)

### CODE REVIEW

File: common/rc [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ##/bin/bash
 # SPDX-License-Identifier: GPL-2.0+
 # Copyright (c) 2000-2006 Silicon Graphics, Inc.  All Rights Reserved.
-
 . common/config
```
Comment (bsbernd) [inline common/rc:4]: Can you keep this line?

File: common/rc [hunk 2]
```diff
@@ -4538,7 +4546,7 @@ init_rc()
 
 	# Sanity check that TEST partition is not mounted at another mount point
 	# or as another fs type
-	_check_mounted_on TEST_DEV $TEST_DEV TEST_DIR $TEST_DIR $FSTYP || exit 1
```
Comment (bsbernd) [inline common/rc:4541]: Why is this changed needed? This should go upstream?

Comment (bsbernd) [reply common/rc:4541]: And what is if FUSE_SUBTYPE is not set?

Comment (d-hans) [reply common/rc:4541]: if FUSE_SUBTYPE is not set then when we pass -o subtype in mount.redfs.passthrough_hp script, check here fails as type is 'redfs.passthrugh_hp' or 'fuse.passthrough_hp'

Comment (bsbernd) [reply common/rc:4541]: Sorry, I still cannot follow you. Why does plain fuse work? It should be an issue without redfs? To me this looks like a non-redfs related change that  needs it is own commit message with an explanation why it is needed. And then it should be send upstream.

File: common/rc (hunk 1 above)
Comment (d-hans) [reply common/rc:4]: done

File: common/rc (hunk 2 above)
Comment (d-hans) [reply common/rc:4541]: plain fuse would not work if -o subtype option is provided in /sbin/mount.fuse.passthrough_hp script. As we are running xfstestuite on passthrough_hp and passthrough_hp requires separate mount script rather than passthrough_ll one, we need subtype option so that mount would uses correct script to run the underlying binary.

**Comment by d-hans** [comment]
raised the commit through another PR therefore closing this
